t/ui/27-plugin_obs_rsync_status_details.t

```diff
@@ -0,0 +1,186 @@
+project_status_url=$url
+EOF
+
+note("Starting fake api server");
+start_server();
+
+my $driver = call_driver(sub { }, {with_gru => 1});
+
+plan skip_all => $OpenQA::SeleniumTest::drivermissing unless $driver;
+
+$driver->find_element_by_class('navbar-brand')->click();
+$driver->find_element_by_link_text('Login')->click();
+
+my %params = (
+    'Proj1'       => ['190703_143010'],
+    'BatchedProj' => ['191216_150610'],
+    'Batch1'      => ['191216_150610', 'BatchedProj'],
+);
+
+sub _wait_helper {
+    my ($element, $test_break) = @_;
+    my $ret;
+    for (my $retries = 0; $retries < 50; $retries = $retries + 1) {
+        $ret = $driver->find_element($element)->get_text();
+        last if &$test_break($ret);
+        sleep(0.1);
+    }
+    return $ret;
+}
+
+foreach my $proj (sort { $b cmp $a } keys %params) {
+    dircopy($home_template, $home);
+    my ($dt, $parent) = @{$params{$proj}};
+    $parent = "" unless $parent;
+
+    $driver->get("/admin/obs_rsync/$parent");
+    my $projfull = $proj;
+    $projfull = "$parent|$proj" if $parent;
+
+    is($driver->find_element("tr#folder_$proj .project")->get_text(), $projfull, "$proj name");
+    ok(index($driver->find_element("tr#folder_$proj .lastsync")->get_text(), $dt) != -1, "$proj last sync");
+    is($driver->find_element("tr#folder_$proj .lastsyncversion")->get_text(), '470.1', "$proj sync version");
+    # at start no project fetches version from obs
+    is($driver->find_element("tr#folder_$proj .obsversion")->get_text(), '', "$proj obs version empty");
+    my $status = $driver->find_element("tr#folder_$proj .dirtystatuscol .dirtystatus")->get_text();
+    ok(index($status, 'dirty') != -1, "$proj dirty status: $status");
+
+    # now request fetching version from obs
+    $driver->find_element("tr#folder_$proj .obsversionupdate")->click();
+    my $obsversion = _wait_helper("tr#folder_$proj .obsversion", sub { shift });
+    is($obsversion, '470.1', "$proj obs version");
+
+    # now we call forget_run_last() and refresh_last_run() and check once again corresponding columns
+    $driver->find_element("tr#folder_$proj .lastsyncforget")->click();
+    $driver->accept_alert;
+
+    my $lastsync = _wait_helper("tr#folder_$proj .lastsync", sub { !shift });
+    ok(index($lastsync, $dt) == -1, "$proj last sync forgotten");
+    is($lastsync, '', "$proj last sync is empty");
+
+    $driver->get("/admin/obs_rsync/$parent");
+    is($driver->find_element("tr#folder_$proj .lastsync")->get_text(), 'no data', "$proj last sync forgotten");
+
+    $driver->find_element("tr#folder_$proj .dirtystatusupdate")->click();
+    # now wait until gru picks the task up
+    my $dirty_status
+      = _wait_helper("tr#folder_$proj .dirtystatuscol .dirtystatus", sub { index(shift, 'dirty') == -1 });
+
+    ok(index($dirty_status, 'dirty') == -1,     "$proj dirty status is not dirty anymore: $dirty_status");
+    ok(index($dirty_status, 'published') != -1, "$proj dirty is published: $dirty_status");
```

Comment:
It's better to use `is`/`isnt` and `like`/`unlike` tests for this, then you get proper diagnostics when the tests fail. Only use `ok` when there is no other choice.

Reply:
I've said it before. I suppose it wouldn't hurt to combine your approach to incorporate `$dirty_status` within the test message with using `is`/`isnt`.

Reply:
Diagnostics string is in the message, I don't see any benefit for using regexp to parse `$dirty_status` here